addons/src/ao_bookmarklist.c

```diff
@@ -62,8 +62,16 @@ struct _AoBookmarkListPrivate
 
 	gint		 search_line;
 	GtkTreeIter	*search_iter;
+
+	guint refresh_idle_source_id;
 };
 
+typedef struct
+{
+	AoBookmarkList *bm;
+	GeanyDocument *document;
```

Comment:
you might want to use Document IDs as you delay the reference to the document.  It probably doesn't matter much in practice though, it's highly unlikely a document is reused during the delay.